scipy/optimize/_minimize.py

```diff
@@ -321,6 +321,9 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
     It should converge to the theoretical solution (1.4 ,1.7).
 
     """
+    x0 = np.asarray(x0)
+    if x0.dtype.kind in np.typecodes["AllInteger"]:
+        x0 = np.asarray(x0, dtype=float)
```

Comment:
On second thought, it's not the same. ``asfarray`` downcasts float96/float128, which can't be what it's supposed to do and is exactly why I added the ``if`` statement above.